Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -40,6 +43,31 @@ The following is a sample attestation result:
 }
 ```
 
+**ctsProfileMatch Vs basicIntegrity**
+
+The SafetyNet Attestation API initially provided a single value called `basicIntegrity` to help developers determine the integrity of a device. As the API evolved, Google introduced a new, stricter check whose results appear in a value called `ctsProfileMatch`, which allows developers to more finely evaluate the devices on which their app is running.
+
+In broad terms, `basicIntegrity` gives you a signal about the general integrity of the device and its API. Many Rooted devices fail `basicIntegrity`, as do emulators, virtual devices, and devices with signs of tampering, such as API hooks.
+
+On the other hand, `ctsProfileMatch` gives you a much stricter signal about the compatibility of the device. Only unmodified devices that have been certified by Google can pass `ctsProfileMatch`. Devices that will fail `ctsProfileMatch` include the following:
+
+- Devices that fail `basicIntegrity`
+- Devices with an unlocked bootloader
+- Devices with a custom system image (custom ROM)
+- Devices for which the manufacturer didn't apply for, or pass, Google certification
+- Devices with a system image built directly from the Android Open Source Program source files
+- Devices with a system image distributed as part of a beta or developer preview program (including the Android Beta Program)
+
+**Recommendations when using** `SafetyNetApi.attest`
+
+-  Create a large (16 bytes or longer) random number on your server using a cryptographically-secure random function so that a malicious user can not reuse a successful attestation result in place of an unsuccessful result
+- Trust APK information (`apkPackageName`, `apkCertificateDigestSha256` and `apkDigestSha256`) only if the value of `ctsProfileMatch` is true.
+- Using a secure connection, the entire JWS response should be sent to your server, for verification. It isn't recommended to perform the verification directly in the app because, in that case, there is no guarantee that the verification logic itself hasn't been modified.
+- The `verify()` method only validates that the JWS message was signed by SafetyNet. It doesn't verify that the payload of the verdict matches your expectations. Also, it has a fixed rate limit of 10,000 requests per day, per project. Hence, use the `verify()` method only for testing during the initial development stage and not in a production scenario instead you should verify the JWS statement at your own server, using a standard cryptographic solutions provided  in the [SafetyNet Verification Samples](https://github.com/googlesamples/android-play-safetynet/tree/master/server/java/src/main/java "Google SafetyNet Sample").

```

Comment:
There is a space too much between 'provided' and 'in' . Next: don't forget to mention that for higher rates google should have been contacted (So more than 10k is possible, but please contact googel for a quota)

Reply:
@commjoen According to Google `verify` method has very strict usage quotas that will not be increased upon request. However, ill talk about Google `SafetyNet.attest` quota in a new bullet.